### PR TITLE
Top Header

### LINKED ISSUE #5: [Frontend] Top Header
# Description
As you can see on the Figma, our goated designer An made a dashboard for the web app. One of the most important things that we have to do when building our frontend is to make components. These will increase modularity and so will allow us to reuse them throughout the app without having to rewrite the same code. You can see an example of a component we made in `src/components/ExampleButton.tsx`.

We’d like you guys to implement the top header as pictured below:

Screen Shot 2024-10-11 at 1 02 24 AM
Screen Shot 2024-10-11 at 1 02 43 AM

As you can see there’s a lot of moving parts but don’t worry, you don’t have to implement the functionality of the buttons yet. You can style the button but don’t worry about the logic when clicking on it. But one thing you should notice is that the admin and volunteer headers are different. For example, there are two cases for the button on the left. Your component should support these cases.

For the notification bell icon, check out Iconify https://icon-sets.iconify.design/ and you can see how we use it in our testing home page.

If this is your first time with Tailwind CSS, don’t worry—their documentation is great. Use the search bar on their website and, for example, look for “height” as shown below:

Screen Shot 2024-10-11 at 1 03 29 AM

The "Class" column is what Tailwind accepts in the “className” of a React component or HTML tag.

You can see an example of how you can use Tailwind in our example button component found in `src/components/ExampleButton.tsx`.

# Testing
You can test your header component by placing it in `src/app/private/admin/home/page.tsx` or `src/app/private/volunteer/home/page.tsx`. One edge case you should definitely check is making sure your component supports a different left button depending on the use case.

Once you’re done with that, feel free to make a PR (Pull Request).

### PR DESCRIPTION
**Pull request by vle04**
# Description

Created a TopHeader,tsx, which is our top header component, and styled the UI.

## Issues

Resolve #5 

## Screenshots

Volunteer header:
Screenshot 2024-10-24 at 1 28 50 PM

Admin header:
Screenshot 2024-10-24 at 1 29 11 PM

## Test

We placed the header in the app's page.tsx in order to see it.
Check the figma to ensure that the design matches.

## Possible Downsides

- Currently we are using a placeholder icon for the profile picture
- In page.tsx, you can change the property passed into TopHeader between admin and volunteer and the UI should change accordingly.
- Note that the buttons do not have any functionality

## Additional Documentations

https://tailwindcss.com

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
 "use client";
 
 import ExampleButton from "@components/ExampleButton";
```
Comment (johnny-t06) [inline src/app/page.tsx:3]: nit: Remove unused imports

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import TopHeader from "@components/TopHeader";
 
 export default function Home() {
   return (
     <div className="flex items-center justify-center h-screen">
       {/* Example Button (ExampleButton.tsx in components folder) */}
-      <ExampleButton buttonText="Add User" />
+      <TopHeader userType="volunteer" />
+      {/* <ExampleButton buttonText="Add User" /> */}
```
Comment (johnny-t06) [inline src/app/page.tsx:11]: It's okay to remove the ExampleButtons now

File: src/components/TopHeader.tsx
Comment (johnny-t06) [inline src/components/TopHeader.tsx:1]: Looks great guys, could we try to make the toggleswitch clickable/toggle-able?

**Review by johnny-t06 — COMMENTED** [review]
The header looks great guys! I left a few comments, you guys can change and then re-request again to take a look.

File: src/components/TopHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  userType: string;
+}
+
+const TopHeader = ({ userType }: TopHeaderProps) => {
+  // different button text and icon depending on if the user is a volunteer or admin
+  const buttonText = userType === "admin" ? "Create new event" : "Check in";
+  const icon = userType === "admin" ? "ic:baseline-plus" : "mdi:checkbox-outline";
+  return (
+    <div className="flex flex-row w-screen border border-black py-8 px-6 place-content-between">
+      <button className="flex flex-row gap-x-2 bg-teal-600 p-2.5 px-3 text-white rounded-md place-items-center">
```
Comment (johnny-t06) [inline src/components/TopHeader.tsx:13]: Also, could we make the header sticky onto the top of the page?

Comment (johnny-t06) [reply src/components/TopHeader.tsx:13]: It would be something like `sticky, top-0` if I remember correctly!

File: src/app/page.tsx (hunk 1 above)
Comment (wkim10) [reply src/app/page.tsx:3]: Like Mr PM said looks really good guys, in general we wanna remove unused imports/variables wherever possible to make the code as clean as possible. It's currently failing to deploy on Vercel because eslint is detecting the unused import while building. You can check if it's building correctly before making a PR by using the "npm run build" command

**Review by wkim10 — APPROVED** [review]
You guys did a great job, the animation especially looks incredible :) I added a few small changes like removing the TopHeader component from the page.tsx file and adding a useState to change the color of the EN and SP text depending on the toggle. I'm gonna merge your branch with main but let me know if you have any questions about my changes, great work guys
